File: Parallel-Csharp/src/Parallel/Entity/Entity.cs
using Parallel.Base;
using Parallel.Vector;

namespace Parallel.Entity;

public class Entity
{
    private int _id;

    public int Id => _id;

    private EntityType _type;

    private EntityFaction _faction;

    private Vector3 _pos;

    private double _maxHealth;

    private double _currentHealth;

    private Entity()
    {
        _id = Global.GenerateEntityId();
        _type = EntityType.Summoned;
        _faction = EntityFaction.Neutral;
        _maxHealth = 10;
        _currentHealth = 10;
        _pos = new Vector3(0, 0, 0);
    }

    public static Entity Create(EntityType type, EntityFaction fac, Vector3 pos, double health)
    {
        var entity = new Entity();
        entity._type = type;
        entity._faction = fac;
        entity._pos = pos;
        entity._maxHealth = health;
        entity._currentHealth = health;
        Global.Instance.GlobalEntityCollection.AddEntity(entity);
        return entity;
    }
    
    public Vector3 Pos
    {
        get => _pos;
        set => _pos = value;
    }
    

    public double MaxHealth
    {
        get => _maxHealth;
        set => _maxHealth = value;
    }

    public double CurrentHealth
    {
        get => _currentHealth;
        set => _currentHealth = value;
    }

    public EntityFaction Faction
    {
        get => _faction;
        set => _faction = value;
    }

    public EntityType Type
    {
        get => _type;
        set => _type = value;
    }

    public static EntityFaction GetOppositeFaction(EntityFaction fac)
    {
        if (fac == EntityFaction.CoolSide)
            return EntityFaction.WarmSide;
        if (fac == EntityFaction.WarmSide)
            return EntityFaction.CoolSide;
        return fac;
    }

    public void TakeDamage(double damageAmont)
    {
        
    }
}

File: Parallel-Java/src/parallel/entity/Entity.java
package parallel.entity;

import parallel.base.Global;
import parallel.vector.Vector3;

public class Entity {
    private int id;

    private EntityType type;

    private EntityFaction faction;

    private Vector3 pos;

    private double maxHealth;

    private double currentHealth;

    private Entity()
    {
        id = Global.GenerateEntityId();
        type = EntityType.SUMMONED;
        faction = EntityFaction.NEUTRAL;
        maxHealth = 10;
        currentHealth = 10;
        pos = new Vector3(0, 0, 0);
    }

    public static Entity create(EntityType type, EntityFaction fac, Vector3 pos, double health)
    {
        var entity = new Entity();
        entity.type = type;
        entity.faction = fac;
        entity.pos = pos;
        entity.maxHealth = health;
        entity.currentHealth = health;
        Global.getInstance().getGlobalEntityCollection().addEntity(entity);
        return entity;
    }

    public int getId() {
        return id;
    }

    public EntityType getType() {
        return type;
    }

    public void setType(EntityType type) {
        this.type = type;
    }

    public EntityFaction getFaction() {
        return faction;
    }

    public void setFaction(EntityFaction faction) {
        this.faction = faction;
    }

    public Vector3 getPos() {
        return pos;
    }

    public void setPos(Vector3 pos) {
        this.pos = pos;
    }

    public double getMaxHealth() {
        return maxHealth;
    }

    public void setMaxHealth(double maxHealth) {
        this.maxHealth = maxHealth;
    }

    public double getCurrentHealth() {
        return currentHealth;
    }

    public void setCurrentHealth(double currentHealth) {
        this.currentHealth = currentHealth;
    }

    public static EntityFaction getOppositeFaction(EntityFaction fac)
    {
        if (fac == EntityFaction.COOL_SIDE)
            return EntityFaction.WARM_SIDE;
        if (fac == EntityFaction.WARM_SIDE)
            return EntityFaction.COOL_SIDE;
        return fac;
    }

    public void takeDamage(double damageAmount) {
    }
}
